### PR TITLE
Support directory for windows

### PR DESCRIPTION
**Pull request by hortigado**

### CODE REVIEW

**Review by tshedor — COMMENTED** [review]
@hortigado I know you mentioned in #377 this is your first PR, and I want to commend you on the bravery to open one. Open source is an intimidating space because it's so exposed - your work is on display for everyone to critique. I really appreciate that you've opened so many issues and taken the step forward to open this PR. Thank you.

However, I apologize for not examining your work more closely yesterday. I should've caught some of my concerns earlier. 

I think the core of what you have here is useful beyond `sqlite_provider` (for example, [the cache manager](https://github.com/GetDutchie/brick/blob/main/packages/brick_offline_first/lib/src/offline_queue/request_sqlite_cache_manager.dart)). A helper function that wraps your logic would be useful in both places. 

```dart
Future<String> _dbName(String fileName) async {
  if (!Platform.isWindows) return fileName;

  final directory = await getApplicationSupportDirectory();
  return '${directory.path}/$fileName';
}
```

A function like this wouldn't live inside the Brick packages because of the Flutter dependency (as convenient as that would be, it breaks the portability). But it should be listed in [the comments of the `sqliteProvider` field of OfflineFirstRepository](https://github.com/GetDutchie/brick/blob/main/packages/brick_sqlite/lib/src/sqlite_provider.dart#L29). This function should also be listed [in the documentation](https://getdutchie.github.io/brick/#/introduction/setup), perhaps under a `Windows Support` header for easy search discovery with a quick example showing how you'd include this function in your repository (maybe as a static method) and where you'd invoke it (on creation of the repository, as the db name for `sqliteProvider` and `offlineQueueManager`).

Again, thank you for opening this PR. This is helpful because I've never run Flutter on Windows and this covers that blind spot. I hope these comments are useful to you and won't discourage you from revising this PR to be documentation focused or from opening a new PR in the future.

File: packages/brick_sqlite/lib/src/sqlite_provider.dart [hunk 1]
```diff
@@ -136,8 +138,17 @@ class SqliteProvider implements Provider<SqliteModel> {
 
   /// Access the latest instantiation of the database [safely](https://github.com/tekartik/sqflite/blob/master/sqflite/doc/opening_db.md#prevent-database-locked-issue).
   @protected
-  Future<Database> getDb() {
-    _openDb ??= databaseFactory.openDatabase(dbName);
+  Future<Database> getDb() async {
```
[the remark this answers is not in the document]
Comment (tshedor) [reply packages/brick_sqlite/lib/src/sqlite_provider.dart:141]: I can do them myself, although you wouldn't get the credit for the code. Is that alright with you if I make a PR with applying these suggestions to the documentation?

[the remark this answers is not in the document]
Comment (hortigado) [reply packages/brick_sqlite/lib/src/sqlite_provider.dart:141]: Yes, better than you, I still have to learn a little more to make a more elaborate PR. Greetings

**Comment by tshedor** [comment]
@hortigado Closing this PR in favor of #387

**Pull request closed without merging**